src/components/vm/overview/vsock.jsx

```diff
@@ -0,0 +1,260 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import cockpit from 'cockpit';
+import { Button } from "@patternfly/react-core/dist/esm/components/Button";
+import { Flex, FlexItem } from "@patternfly/react-core/dist/esm/layouts/Flex";
+import { Form, FormGroup } from "@patternfly/react-core/dist/esm/components/Form";
+import { NumberInput } from "@patternfly/react-core/dist/esm/components/NumberInput";
+import { Modal } from "@patternfly/react-core/dist/esm/components/Modal";
+import { Checkbox } from "@patternfly/react-core/dist/esm/components/Checkbox";
+import { useDialogs } from 'dialogs.jsx';
+import { Remarkable } from "remarkable";
+
+import { ModalError } from 'cockpit-components-inline-notification.jsx';
+import { FormHelper } from "cockpit-components-form-helper.jsx";
+import { domainRemoveVsock, domainSetVsock } from "../../../libvirtApi/domain.js";
+import { NeedsShutdownAlert, NeedsShutdownTooltip } from "../../common/needsShutdown.jsx";
+
+import "./vsock.scss";
+
+const _ = cockpit.gettext;
+
+const ASSIGN_AUTOMATICALLY = _("Assign automatically");
+// There  are  several  reserved  addresses:
+// VMADDR_CID_ANY  (-1U)  means  any  address  for  binding;
+// VMADDR_CID_HYPERVISOR (0) is reserved for services built into the hypervisor;
+// VMADDR_CID_LOCAL (1) is the well-known address for local communication (loopback);
+// VMADDR_CID_HOST (2) is the well-known address of the host.
+const MIN_VSOCK_CID = 3;
+
+function getVsockUsageMessage(vmName, connectionName, vms, auto, address) {
+    if (auto)
+        return;
+
+    const remarkable = new Remarkable();
+
+    const vmsUsingCIDAddress = vms
+            .filter(vm => !(vmName === vm.name && connectionName === vm.connectionName) && vm.vsock.cid.auto === "no" && vm.vsock.cid.address === String(address))
+            .map(vm => vm.name);
+    if (vmsUsingCIDAddress.length === 0)
+        return;
+
+    const vmsUsingCIDAddressString = vmsUsingCIDAddress.join(", ");
+
+    return (
+        <span dangerouslySetInnerHTML={{
+            __html: remarkable.render(cockpit.format(_("Identifier in use by **$0** VMs with an identical identifier cannot run at the same time."), vmsUsingCIDAddressString))
```

Comment:
I think this is missing a period to separate spaces.

```suggestion
            __html: remarkable.render(cockpit.format(_("Identifier in use by **$0**. VMs with an identical identifier cannot run at the same time."), vmsUsingCIDAddressString))
```

Reply:
So this addresses my [earlier review comment](https://github.com/cockpit-project/cockpit-machines/pull/969#pullrequestreview-1421755844), but this also seems brittle to me: If translations manage to do anything wrong to these `**`, put them in unbalanced or an unexpected order, or translators try to do something else with them, or happen to use any more "magic" markdown character, this is going to be a mess and crash the page in the worst case.

If you want to do this, please check `git grep fmt_to_fragments` in cockpit -- that is in pkg/lib, you can use it in c-machines. This keeps the markup outside of the translatable bits, and is both much more robust as well as also more efficient than rendering through markdown.

Reply:
Fixed and used fmt_to_fragments